### PR TITLE
Initialize cobra-cli repo

### PR DESCRIPTION
**Pull request by liggitt**
This PR sets up various housekeeping things for the new repo:
* Copies existing license/contributing/conduct/CI root files from spf13/cobra
* Adds copies of make targets from spf13/cobra, adjust CI config
* Rename packages and go.mod to cobra-cli
* Tidy go module

Because this repo placed the binary at the root, it makes the binary name change to `cobra-cli`. Before merging any changes into this repo, we should make sure that's what we want, since that ripples to docs and usage, not just the install command (asked in https://github.com/spf13/cobra/issues/1597#issuecomment-1034462012)

If we want to preserve the `cobra` binary, we should reinitialize it with the history placed under a `github.com/spf13/cobra-cli/cobra` subpackage.

Preview of the configured CI runs passing at https://github.com/liggitt/cobra-cli/actions?query=branch%3Ainit

### CODE REVIEW

**Comment by liggitt** [comment]
ack

**Comment by liggitt** [comment]
sounds like the consensus was to keep the `cobra-cli` name, so this is ready, content-wise. moving out of draft, but will hold for merge until @spf13 gets resolution on the DCO/CLA thing.

**Review by jpmcb — COMMENTED** [review]
Also looks good to me!

After this merges and the dust has settled a bit, we can touch up the readme, add labels to this repo, and do other maintence bits as needed

**Comment by liggitt** [comment]
> I'm going to bring this up with someone with deeper expertise in the legal field of open source and see if I can get more guidance. If this can wait for a few days that would be best.

@spf13, any update on this bit?

**Comment by spf13** [comment]
Meeting today on it.

**Comment by spf13** [comment]
I have an update which you can read https://github.com/spf13/cobra/issues/1555#issuecomment-1043261602 

As a result of these findings, I'll go ahead and setup CLA Assistant on this repo too.

**Comment by spf13** [comment]
I've setup CLA Assistant but I believe that because this PR predated it it's not being required. It might require to resubmit the PR to trigger it. @liggitt would that be too much trouble. I'm not sure what alternatives we have.

**Comment by spf13** [comment]
That worked!! Thanks @liggitt

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      fail-fast: false
+      matrix:
+        platform:
+        - ubuntu
+        - macOS
+        go:
+        - 14
+        - 15
+        - 16
+        - 17
```
Comment (bep) [inline .github/workflows/test.yml:38]: I would recommend that you also include Windows in the matrix and reduce the numbers by cutting one or two Go versions.

Comment (jpmcb) [reply .github/workflows/test.yml:38]: We do test against windows here: 

https://github.com/spf13/cobra-cli/blob/7c6e26ac0939450323b7dd4acf8d8280dde3879e/.github/workflows/test.yml#L66

And yes, will remove Go 14 and 15.

**Pull request merged**